metrics/internal/collectors/cluster-advance-feature-use.go

```diff
@@ -77,102 +69,93 @@ func (c *ClusterAdvanceFeatureCollector) Describe(ch chan<- *prometheus.Desc) {
 
 // Collect implements prometheus.Collector interface
 func (c *ClusterAdvanceFeatureCollector) Collect(ch chan<- prometheus.Metric) {
+	// advancedFeature will be set to
+	// '1' if any of the cluster is using an advanced feature
+	// or else it will be set to '0'.
+	advancedFeature := 0
+
 	cephClusterLister := cephv1listers.NewCephClusterLister(c.Informer.GetIndexer())
 	cephClusters := getAllCephClusters(cephClusterLister, c.AllowedNamespaces)
 	if len(cephClusters) > 0 {
-		c.mapAdvanceFeatureUseFromCephClusters(cephClusters)
+		advancedFeature = c.advancedFeatureFromCephClusters(cephClusters)
+	}
+	if advancedFeature > 0 {
+		c.collectAdvancedFeatureUse(ch, advancedFeature)
+		return
 	}
 
 	cephObjectStoreLister := cephv1listers.NewCephObjectStoreLister(c.Informer.GetIndexer())
 	cephObjectStores := getAllObjectStores(cephObjectStoreLister, c.AllowedNamespaces)
 	if len(cephObjectStores) > 0 {
-		c.mapAdvanceFeatureUseFromCephObjectStores(cephObjectStores)
+		advancedFeature = c.advancedFeatureFromCephObjectStores(cephObjectStores)
+	}
+	if advancedFeature > 0 {
+		c.collectAdvancedFeatureUse(ch, advancedFeature)
+		return
 	}
 
 	storageClassLister := storagev1listers.NewStorageClassLister(c.Informer.GetIndexer())
 	storageClasses := getAllStorageClasses(storageClassLister, c.AllowedNamespaces)
 	if len(storageClasses) > 0 {
-		c.mapAdvanceFeatureUseFromStorageClasses(storageClasses)
+		advancedFeature = c.advancedFeatureFromStorageClasses(storageClasses)
+	}
+	if advancedFeature > 0 {
+		c.collectAdvancedFeatureUse(ch, advancedFeature)
+		return
 	}
 
-	c.collectAdvancedFeatureUse(ch)
-}
-
-func (c *ClusterAdvanceFeatureCollector) findValidKeyInNamespace(namespace string) string {
-	var validKey string
-	for namespaceNameKey := range c.advancedFeatureMap {
-		// namespaceNameKey is in the format: <namespace>/<clusterName>
-		if strings.HasPrefix(namespaceNameKey, fmt.Sprint(namespace, "/")) {
-			validKey = namespaceNameKey
-			break
-		}
+	cephRBDMirrorLister := cephv1listers.NewCephRBDMirrorLister(c.Informer.GetIndexer())
+	cephRBDMirrors := getAllRBDMirrors(cephRBDMirrorLister, c.AllowedNamespaces)
+	if len(cephRBDMirrors) > 0 {
+		advancedFeature = c.advancedFeatureFromCephRBDMirrors(cephRBDMirrors)
 	}
-	return validKey
+
+	c.collectAdvancedFeatureUse(ch, advancedFeature)
```

Comment:
Don't we need to check if `advancedFeature > 0` here?

Reply:
Last `collectAdvancedFeatureUse()` call should be called unconditionally so that we should collect (whether) ZERO or ONE and send the metric. This is needed if customer resets a special feature then metric should reflect that (or if customer is not using any advanced feature).